### PR TITLE
feat(ui5-li-notification): implement new design

### PR DESCRIPTION
**Pull request by LidiyaGeorgieva**
There is new design for the Notifications.
In the current pull request we implement the changes for the Notification Item:
- new slot "menu" is introduced (to replace the ui5-notification-action)
- ui5-notification-action is deprecated (still will be visualized, but will be completely removed in future)
- if there is only one action it will go on the overflow (before it was rendered as separate button)
- samples are updated

BGSOFUIRODOPI-3226

- "priority" property is depricated (but stil works)
- "statusIndicator" prerty is added. It will replace the "priority" property
- "indicateImportance" property is added. It is for visualizing "Importance" label

BGSOFUIRODOPI-3227

### CODE REVIEW

**Review by BorisDafov — CHANGES_REQUESTED** [review]

**Review by TeodorTaushanov — CHANGES_REQUESTED** [review]

File: packages/fiori/test/specs/NotificationList.spec.js [hunk 1]
```diff
@@ -226,16 +226,34 @@ describe("Notification List Item Tests", () => {
...
 	it("tests Menu (actions / '...') button ACC attributes", async () => {
 		const firstItem = await browser.$("#nli1");
-		const menuButton = await firstItem.shadow$(".ui5-nli-menu-btn");
+		const menuButton = await firstItem.shadow$(".ui5-nli-menu-btn").shadow$("button");
+		const thirdItem = await browser.$("#nli3");
+		const btnListItemMenu3 = await thirdItem.shadow$(".ui5-nli-menu-btn");
 
 		// assert
-		assert.strictEqual(await menuButton.getAttribute("tooltip"), 'Actions', "The tooltip text is correct.");
+		assert.strictEqual(await menuButton.getAttribute("title"), 'Actions', "The tooltip text is correct.");
```
Comment (BorisDafov) [inline packages/fiori/test/specs/NotificationList.spec.js:234]: Pls double check - the tooltip or the title.

**Review by BorisDafov — COMMENTED** [review]
From UA perspective, I approve the PR

File: packages/fiori/test/specs/NotificationList.spec.js (hunk 1 above)
Comment (LidiyaGeorgieva) [reply packages/fiori/test/specs/NotificationList.spec.js:234]: the 'tooltip' is the property, which adds 'title' attribute to the <button>
It depends which element is checked.

**Comment by LidiyaGeorgieva** [comment]
> Looks good overall. Small thing:
> 
> 1. in NotificationListGroupItem.html, when the notification are really small the space between the header text and content is not enough

Done.

File: packages/fiori/test/pages/NotificationListItem.html [hunk 2]
```diff
@@ -17,26 +17,72 @@
 	<ui5-shellbar id="shellbar" primary-title="Product Title" show-notifications>
 	</ui5-shellbar>
 
-	<ui5-list id="notificationList" header-text="Notifications titleText and content 'truncates'">
+	<ui5-list id="notificationListWithMenu" header-text="Notifications that use Menu for actions as per the new design (2024)">
```
Comment (BorisDafov) [inline packages/fiori/test/pages/NotificationListItem.html:20]: I suggest to skip the additional information (soon it wouldn't be new):
Notifications that use Menu for actions

File: packages/fiori/test/pages/NotificationListItem.html [hunk 3]
```diff
@@ -17,26 +17,72 @@
...
+
+			<span slot="footnotes">John Doe</span>
+			<span slot="footnotes">2 Days</span>
+			<span slot="footnotes">Other stuff</span>
+		</ui5-li-notification>
+
+		<ui5-li-notification
+			show-close
+			state="Information"
+			title-text="When 'state' (new) is set to 'Information' we need to see it"
```
Comment (BorisDafov) [inline packages/fiori/test/pages/NotificationListItem.html:67]: I suggest:
When 'state' is set to 'Information' we need to see it

File: packages/fiori/test/pages/NotificationListItem.html [hunk 4]
```diff
@@ -45,32 +91,75 @@
...
+				<img src="./img/man_avatar_1.png" />
+			</ui5-avatar>
+
+			<span slot="footnotes">John Doe</span>
+			<span slot="footnotes">2 Days</span>
+			<span slot="footnotes">Other stuff</span>
+		</ui5-li-notification>
+	</ui5-list>
+
+	<ui5-list id="notificationList" header-text="Notifications titleText and content 'truncates'">
```
Comment (BorisDafov) [inline packages/fiori/test/pages/NotificationListItem.html:109]: I suggest:
Notifications 'titleText' and content is truncated

**Review by BorisDafov — CHANGES_REQUESTED** [review]

**Review by TeodorTaushanov — APPROVED** [review]

**Pull request merged**